src/domain.py

```diff
@@ -707,6 +709,14 @@ def _apply(self):
                     domain=self.entity,
                     other_app=app_map(raw=True)[self.entity]["/"]["id"],
                 )
+        if (
+            "recovery_password" in self.future_values
+            and self.future_values["recovery_password"]
+            != self.values["recovery_password"]
+        ):
+            domain_dyndns_set_recovery_password(
+                self.entity, self.future_values["recovery_password"]
+            )
```

Comment:
Hmmm it looks like in the current code, the recovery password is effectively saved to the file, but we dont want this. It's only meant to be hashed and sent to the API and stored (hashed *again*) on the client side

I don't think we want to validate that the `future_values` is different from the current(?) `values` ...

this is similar to the global settings panel where people can now change their root password but we don't want to save it in cleartext etc ... 

cf https://github.com/YunoHost/yunohost/blob/dev/src/settings.py#L216-L227

I got a bit paranoid in that bit of code, also because there can be some confusion between `new_values`, `future_values` etc...